File: app/src/main/java/com/druger/aboutwork/adapters/BaseViewHolder.java
package com.druger.aboutwork.adapters;

import android.view.View;

import androidx.annotation.IdRes;
import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by druger on 13.08.2017.
 */

public abstract class BaseViewHolder extends RecyclerView.ViewHolder {

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    @SuppressWarnings("unchecked")
    protected <T extends View> T bindView(@IdRes int id) {
        return (T) itemView.findViewById(id);
    }
}

File: app/src/main/java/com/druger/aboutwork/adapters/BaseViewHolder.kt
package com.druger.aboutwork.adapters

import android.view.View

import androidx.annotation.IdRes
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by druger on 13.08.2017.
 */

abstract class BaseViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    protected fun <T : View> bindView(@IdRes id: Int): T {
        return itemView.findViewById<View>(id) as T
    }
}
